File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if (intervals.empty()) return {};
        sort(intervals.begin(),intervals.end());
        vector<vector<int>>ans;
        ans.push_back(intervals[0]);
        int j=0;
        for(int i=1; i<intervals.size();i++){
            if(ans[j][1]<intervals[i][0]){
                j++;ans.push_back(intervals[i]);
            }else{
                ans[j][1]= max(ans[j][1],intervals[i][1]);
            }
        }
        return ans;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, inr: List[List[int]]) -> List[List[int]]:
        inr.sort()
        ans = [inr[0]]
        for it in inr:
            if ans[-1][1] < it[0]: ans.append(it)
            else: ans[-1][1] = max(ans[-1][1], it[1])
        return ans
